production/helm/templates/promtail/daemonset.yaml

```diff
@@ -51,11 +53,10 @@ spec:
               fieldRef:
                 fieldPath: spec.nodeName
           ports:
-            - containerPort: 80
+            - containerPort: {{ .Values.promtail.port }}
```

Comment:
This would mess with our assumptions of Prometheus scraping this daemonset under :80. 
This is going to be hard to square. WDYT @tomwilkie ?

Reply:
As I understand, you guys use ksonnet in production and not the helm chart. The ksonnet remains unchanged. Also, everything in this PR is configurable so you can always provide a custom values.yaml that overrides these options to make it as before.

Reply:
promtail port is a container pod, as i understand,  there is no conflict with other services, config this value is meaningless.

Reply:
this port is just container pod, config as another val is meaningless since there is no possible about conflict with other services.

Reply:
It's not meaningless, you can't run as non-root with ports < 1024. Prometheus also scrapes this port directly since promtail doesn't have a service.

Reply:
running non-root is preferable.

Thinking about it though, I'm not sure it can do that. Doesn't it need root to be able to read all the logs? If root is required, keeping it to port 80 doesn't hurt then.

Reply:
Root isn't required, it can read any logs that are mounted via `promtail.volumes`. I've of course tested this locally and it works.

Reply:
How are the logs from the host not root only readable?

Reply:
ls -l /var/log/pods/a5560f9e-03b2-11e9-ab60-0cc47a18625e/main/6.log
-rw-r----- 1 root root 0 Feb 19 15:27 /var/log/pods/a5560f9e-03b2-11e9-ab60-0cc47a18625e/main/6.log

On my system they seem to be.

Reply:
I have the same permissions. It works because of the group read permission and the fact that RunAsGroup feature gate is not enabled by default until 1.14. So it was effectively gid=0. But that's brittle, so I've switched back to RunAsUser=0 by default. I will be setting RunAsUser=10001 and RunAsGroup=0 in my custom values since that works in my cluster.

Reply:
Ah, ok. That explains it. Thanks. :)

Reply:
> It's not meaningless, you can't run as non-root with ports < 1024. Prometheus also scrapes this port directly since promtail doesn't have a service.

I mean port change, reduced privileges is good.

Reply:
If nobody has any objections, we'll go ahead with using `.Values.promtail.port` here.